src/server/routes/revision.js

```diff
@@ -178,52 +178,53 @@ function formatEditionGroupChange(change) {
 
 function diffRevisionsWithParents(orm, entityRevisions, entityType) {
 	// entityRevisions - collection of *entityType*_revisions matching id
-	return Promise.all(entityRevisions.map(
+	const promises = entityRevisions.map(
 		async (revision) => {
 			const dataId = revision.get('dataId');
 			const revisionEntity = revision.related('entity');
 			const entityBBID = revisionEntity.get('bbid');
 			const entity = await orm.func.entity.getEntity(orm, entityType, entityBBID);
 			const isEntityDeleted = !entity.dataId;
-			return revision.parent()
-				.then(
-					(parent) => {
-						let isNew = false;
-						const isDeletion = !dataId;
-						if (!parent) {
-							isNew = Boolean(dataId);
-						}
-						return makePromiseFromObject({
-							changes: revision.diff(parent),
-							entity: revisionEntity,
-							entityAlias: dataId ?
-								revision.related('data').fetch({require: false, withRelated: ['aliasSet.defaultAlias', 'aliasSet.aliases']}) :
-								orm.func.entity.getEntityParentAlias(
-									orm, entityType, revision.get('bbid')
-								),
-							isDeletion,
-							isEntityDeleted,
-							isNew,
-							revision
-						});
-					},
-					// If calling .parent() is rejected (no parent rev), we still want to go ahead without the parent
-					() => makePromiseFromObject({
-						changes: revision.diff(null),
-						entity: revisionEntity,
-						entityAlias: revision.get('dataId') ?
-							revision.related('data').fetch({require: false, withRelated: ['aliasSet.defaultAlias', 'aliasSet.aliases']}) :
-							orm.func.entity.getEntityParentAlias(
-								orm, entityType, revision.get('bbid')
-							),
-						isDeletion: !dataId,
-						isEntityDeleted,
-						isNew: Boolean(dataId),
-						revision
-					})
+			try {
+				const parent = await revision.parent();
+				let isNew = false;
+				const isDeletion = !dataId;
+				if (!parent) {
+					isNew = Boolean(dataId);
+				}
+				return makePromiseFromObject({
+					changes: revision.diff(parent),
+					entity: revisionEntity,
+					entityAlias: dataId ?
+						await revision.related('data').fetch({require: false, withRelated: ['aliasSet.defaultAlias', 'aliasSet.aliases']}) :
```

Comment:
Here I believe we don't want to `await` the promise, as that is what `makePromiseFromObject` is used for.
It returns a Promise resolving to an object with each value resolved (if the value was a promise).